code/datums/brain_damage/creepy_trauma.dm

```diff
@@ -125,15 +126,10 @@
 			viable_minds += Player.mind
 	for(var/datum/mind/possible_target in viable_minds)
 		if(possible_target != owner && ishuman(possible_target.current))
-			if(HAS_TRAIT(owner, TRAIT_FAN_MIME) && possible_target.assigned_role == "Mime")
-				special_pool += possible_target.current
-				continue
-			if(HAS_TRAIT(owner, TRAIT_FAN_CLOWN) && possible_target.assigned_role == "Clown")
-				special_pool += possible_target.current
-				continue
-			if(HAS_TRAIT(owner, TRAIT_SPIRITUAL) && possible_target.assigned_role == "Chaplain")
-				special_pool += possible_target.current
-				continue
+			for(var/trait in trait_obsessions)
+				if(HAS_TRAIT(owner, trait) && possible_target.assigned_role == trait_obsessions[trait])
+					special_pool += possible_target.current
+					break
```

Comment:
This is why I said job -> trait, and not trait -> job.

If you do job -> trait, you can save the loop and just check:

```dm
if (trait_obsessions[job] != null && HAS_TRAIT(owner, trait_obsessions[job]))
```